src/main/java/frc/robot/commands/StateMachine.java

```diff
@@ -0,0 +1,115 @@
+
+        public final IntakeState intakeState;
+        public final ManipulatorState manipulatorState;
+        public final ElevatorState elevatorState;
+
+        private RobotState(IntakeState intakeState, ManipulatorState manipulatorState, ElevatorState elevatorState) {
+            this.intakeState = intakeState;
+            this.manipulatorState = manipulatorState;
+            this.elevatorState = elevatorState;
+        }
+    }
+
+    private RobotState state;
+    private final Intake intake;
+    private final CoralManipulator manipulator;
+    private final Elevator elevator;
+
+    public StateMachine(Intake intake, CoralManipulator manipulator, Elevator elevator) {
+        this.intake = intake;
+        this.manipulator = manipulator;
+        this.elevator = elevator;
+        this.state = RobotState.IDLE;
+
+        stateTable = NetworkTableInstance.getDefault().getTable("StateMachine");
+        currentStatePub = stateTable.getStringTopic("CurrentState").publish();
+        intakeStatePub = stateTable.getStringTopic("SubsystemStates/Intake").publish();
+        manipulatorStatePub = stateTable.getStringTopic("SubsystemStates/Manipulator").publish();
+        elevatorStatePub = stateTable.getStringTopic("SubsystemStates/Elevator").publish();
+
+        publishStates();
+    }
+
+    @Override
+    public void execute() {
+        publishStates();
+
+        // returning to idle after scoring
+        if (isScoreState(state) && !manipulator.hasCoral()) {
+            Commands.waitSeconds(0.5)
+                   .andThen(() -> scheduleNewState(RobotState.IDLE))
+                   .schedule();
+        }
+    }
+
+    public void scheduleNewState(RobotState newState) {
+        changeState(newState).schedule();
+    }
+
+    public Command changeState(RobotState newState) {
+        return Commands.sequence(
+            Commands.runOnce(() -> {
+                state = newState;
+                System.out.println(newState.toString() + " scheduled");
+                publishStates();
+            }),
+            // first handle elevator state to avoid collisions
+            elevator.changeState(newState.elevatorState),
+            // next, change manipulator and intake states in parallel
+            Commands.parallel(
+                manipulator.changeState(newState.manipulatorState),
+                intake.changeState(newState.intakeState)
+            )
+        );
+    }
+
+    private boolean isScoreState(RobotState state) {
+        return state == RobotState.SCORE_L1 || 
+               state == RobotState.SCORE_L2 || 
+               state == RobotState.SCORE_L3 || 
+               state == RobotState.SCORE_L4;
+    }
+
+    private void publishStates() {
+        currentStatePub.set(state.toString());
+        intakeStatePub.set(state.intakeState.toString());
+        manipulatorStatePub.set(state.manipulatorState.toString());
+        elevatorStatePub.set(state.elevatorState.toString());
```

Comment:
you will need to do some better coordination here. For example, we don't want to publish the manipulator state for scoring until the elevator is in position